src/core/tranche/quant.sol

```diff
@@ -17,73 +17,90 @@ pragma solidity >=0.4.24;
 
 import "ds-note/note.sol";
 
+contract ReserveLike {
+   function balance() public returns(uint);
+}
+
 // Quant
-// Keeps track of the tranche debt. Manages iBorrow & its calculation.
+// Keeps track of the tranche debt / expected tranche returns. Manages borrowRate & its calculation.
 contract Quant is DSNote {
     // --- Auth ---
     mapping (address => uint) public wards;
     function rely(address usr) public auth note { wards[usr] = 1; }
     function deny(address usr) public auth note { wards[usr] = 0; }
     modifier auth { require(wards[msg.sender] == 1); _; }
 
+    // --- Data ---
+    ReserveLike public reserve;
+
     struct Fee {
         uint chi;
         uint speed; // Accumulation per second
         uint48 rho; // Last time the rate was accumulated
     }
     
-    Fee public iBorrow;
+    Fee public borrowRate;
     uint public debt;
+    uint public supplyRate;
+    bool public supplyRateFixed;
     
-    constructor() public {
+    constructor(address reserve_) public {
+        reserve = ReserveLike(reserve_);
         wards[msg.sender] = 1;
-        iBorrow.chi = ONE;
-        iBorrow.speed = ONE;
+        borrowRate.chi = ONE;
+        borrowRate.speed = ONE;
+        supplyRate = ONE;
     }
 
     function file(bytes32 what, uint speed_) public note auth {
-         if (what == "iborrow") {
+         if (what ==  "borrowrate") {
             drip();
-            iBorrow.speed = speed_;
+            borrowRate.speed = speed_;
         }
-    }
-
-    function getSpeed() public returns(uint) {
-        return iBorrow.speed;
-    }
-
-    function UpdateIBorrow(uint supplySpeed, uint reserve) public note auth {
-        require (supplySpeed > 0);
-        if (now >= iBorrow.rho) {
+        else if (what ==  "supplyrate") {
             drip();
+            supplyRate = speed_;
         }
-        uint supplySpeed_ = sub(supplySpeed, ONE);
-        uint ratio = rdiv(add(reserve, debt), debt); 
-        iBorrow.speed = add(rmul(ratio, supplySpeed_), ONE);
     }
 
-    function drip() public note auth{
-        if (now >= iBorrow.rho) {
-            (uint latest, , uint wad) = compounding();
-            iBorrow.chi = latest;
-            iBorrow.rho = uint48(now);
-            debt = add(debt, wad);   
+    function setFixedSupplyRate(bool fixed_) public note auth {
+        supplyRateFixed = fixed_;
+    }
+
+    function updateBorrowRate() public note auth {
+        if (supplyRateFixed && supplyRate > 0) {
+            if (now >= borrowRate.rho) {
+                drip();
+            }
+            uint balance = reserve.balance();
+            uint supplyRate_ = sub(supplyRate, ONE);
+            uint ratio = rdiv(add(balance, debt), debt); 
+            borrowRate.speed = add(rmul(ratio, supplyRate_), ONE);
```

Comment:
In general the formular is correct but don't we run into our decimal number issue here.
assuming supplyRate is something like `1000000564701133626865910626` => 10^27.
`1000000564701133626865910626` - ONE would result in `564701133626865910626`.

Didn't we come up already with a solution for that?